### PR TITLE
add options for balsamic

### PR DESCRIPTION
**Pull request by patrikgrenfeldt**
This PR adds options to start balsamic with:

**How to test**:
1. install on stage of the hasta machine: `bash servers/resources/hasta.scilifelab.se/update-cg-stage.sh add-balsamic-config-options
1. activate stage: `us`

**How to test default options**:
- [x] run following command: `cg analysis balsamic firmshiner --dry`

**Expected outcome**:
- [x] `cg` should have the --singularity specified with a value of "/home/proj/production/cancer/BALSAMIC/BALSAMIC/containers/BALSAMIC.3.2.0" in the command output that we start balsamic with
- [x] `cg` should have the --umi-trim-length specified with a value of 5 in the command output 

**How to test option --singularity **:
- [x] run following command: `cg analysis balsamic firmshiner --dry --singularity SINGULARITYPATH`

**Expected outcome**:
- [x] `cg` should have the --singularity specified in the command output with the value SINGULARITYPATH that we start balsamic with

**How to test option --umi-trim-length**:
- [x] run following command: `cg analysis balsamic firmshiner --dry --umi-trim-length`

**Expected outcome**:
- [x] `cg` should have the --umi-trim-length specified in the command output that we start balsamic with

**How to test option --quality-trim**:
- [x] run following command: `cg analysis balsamic firmshiner --dry --quality-trim`

**Expected outcome**:
- [x] `cg` should have the --quality-trim specified in the command output that we start balsamic with

**How to test option --adapter-trim**:
- [x] run following command: `cg analysis balsamic config firmshiner --dry --adapter-trim`

**Expected outcome**:
- [x] `cg` should have the --adapter-trim specified in the command output that we start balsamic with

**How to test option --umi**:
- [x] run following command: `cg analysis balsamic config firmshiner --dry --umi`

**Expected outcome**:
- [x] `cg` should have the --umi specified in the command output that we start balsamic with

**How to test options with balsamic**:
- [x] run following command: `cg analysis balsamic config firmshiner --quality-trim --adapter-trim --umi`

**Expected outcome**:
- [x] `balsamic` should not complain on any of the options given 

Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by @ingkebil 
- [x] tests executed by @emiliaol 
- [x] "Merge and deploy" approved by @ingkebil 
Thanks for filling in who performed the code review and the test!

This is |minor|**version bump** because options should have been added in a way that it starts balsamic just as before but it is possible to give new options as well

### CODE REVIEW

File: cg/cli/balsamic.py [hunk 1]
```diff
@@ -25,9 +26,17 @@ def balsamic(context):
 @balsamic.command()
 @click.option('-d', '--dry', is_flag=True, help='print config to console')
 @click.option('--target-bed', required=False, help='Optional')
+@click.option('--singularity', default="/home/proj/production/cancer/cases/github_issue_analysis"
```
Comment (emiliaol) [inline cg/cli/balsamic.py]: There is also a better path to the container: `/home/proj/production/cancer/BALSAMIC/BALSAMIC/containers/BALSAMIC.3.2.0`

Comment (hassanfa) [reply cg/cli/balsamic.py]: We already discussed this. But here is the issue I created on cg regarding above changes: https://github.com/Clinical-Genomics/cg/issues/453

**Comment by hassanfa** [comment]
Emilia, here is the issue I mentioned earlier: https://github.com/Clinical-Genomics/cg/issues/453

**Comment by emiliaol** [comment]
> Emilia, here is the issue I mentioned earlier: #453

👍

File: cg/cli/balsamic.py (hunk 1 above)
Comment (patrikgrenfeldt) [reply cg/cli/balsamic.py]: Thanx!

**Comment by patrikgrenfeldt** [comment]
@emiliaol I created some test case, could you test them out?

File: cg/cli/balsamic.py (hunk 1 above)
Comment (ingkebil) [reply cg/cli/balsamic.py]: deosn't really fit well to put such specific information as a default in the CLI. Information like this should be moved to the config

File: cg/cli/balsamic.py [hunk 2]
```diff
@@ -25,9 +26,16 @@ def balsamic(context):
 @balsamic.command()
 @click.option('-d', '--dry', is_flag=True, help='print config to console')
 @click.option('--target-bed', required=False, help='Optional')
+@click.option('--singularity', default="/home/proj/production/cancer/BALSAMIC/BALSAMIC/containers"
```
Comment (ingkebil) [inline cg/cli/balsamic.py]: Can we move this default to the config instead?

**Comment by ingkebil** [comment]
Yeah, we need to find a better way of handling so many options. Otherwise, looks good.

File: cg/cli/balsamic.py (hunk 2 above)
Comment (emiliaol) [reply cg/cli/balsamic.py]: We will. We should also add the reference and environment to the cg config. But separate PR to be done for that.

**Comment by emiliaol** [comment]
Default options tested:
Skärmavbild 2019-10-22 kl  13 46 24 👍

**Comment by emiliaol** [comment]
test option singularity:
Skärmavbild 2019-10-22 kl  13 49 21
👍

**Comment by emiliaol** [comment]
test option --umi-trim-length:
Skärmavbild 2019-10-22 kl  13 51 52
👍

**Comment by emiliaol** [comment]
option --quality-trim:
Skärmavbild 2019-10-22 kl  13 53 22

👍

**Comment by emiliaol** [comment]
test option --adapter-trim:
Skärmavbild 2019-10-22 kl  13 57 32
👎 Option has _ instead of -

**Comment by emiliaol** [comment]
test --umi option:
Skärmavbild 2019-10-22 kl  13 59 08
👍

**Comment by emiliaol** [comment]
re-tested --umi-trim-length option:
Skärmavbild 2019-10-22 kl  14 22 57
👍

**Comment by emiliaol** [comment]
re-tested --quality-trim option:
Skärmavbild 2019-10-22 kl  14 25 11
👍

**Comment by emiliaol** [comment]
re-testing --adapter-trim:
Skärmavbild 2019-10-22 kl  14 27 51
👍

**Comment by emiliaol** [comment]
testing the options without dry-run:
Skärmavbild 2019-10-22 kl  14 43 50
👍 
There are some errors but they are balsamic and not cg related. The config was created according to specifications.

**Comment by emiliaol** [comment]
@ingkebil Tested and approved. Time to deploy? 😃

**Comment by ingkebil** [comment]
Approved. I'll push when Hasta is back

**Comment by ingkebil** [comment]
deployed

**Comment by emiliaol** [comment]
Great, but the cg config is not updated?

**Comment by ingkebil** [comment]
This PR doesn't mention updating the cg config. Could you elaborate on what the update should consist of?

**Comment by ingkebil** [comment]
Alright, I know what you mean. @patrikgrenfeldt can you Pr those changes as well?

**Pull request merged**